### PR TITLE
Add utf-8 filter-validator

### PR DESCRIPTION
**Pull request by mmorhun**
### What does this PR do?

Adds utf-8 byte stream validator which replaces invalid bytes in the stream with utf-8 placeholder character.
Also it handles the case when an utf-8 character is separated between two data portions. In such case bytes which could mean an utf-8 character is held until the next data portion comes.

Adds tests for provided filter.

### Related issues

Fixes https://github.com/eclipse/che-machine-exec/issues/20

### CODE REVIEW

File: api/model/rune_reader.go [hunk 1]
```diff
@@ -13,33 +13,153 @@
...
+			result.WriteRune(char)
+			processedBytes += charLen
+		}
+	}
+}
+
+// handleLastBytes processes from 1 to 3 last bytes of given data
+// which aren't complete utf-8 character (but may contain such).
+// Returns data (placeholder characters), if any, which should be returned to data stream
+// and saves remainder, if any, for the next datat part.
```
Comment (AndrienkoAleksandr) [inline api/model/rune_reader.go]: Typo: datat -> data

**Comment by AndrienkoAleksandr** [comment]
@mmorhun It would be nice to move rune-reader from model package to the separated one.

File: api/model/rune_reader.go (hunk 1 above)
Comment (mmorhun) [reply api/model/rune_reader.go]: Thank you, will fix

**Comment by mmorhun** [comment]
@AndrienkoAleksandr done

**Comment by AndrienkoAleksandr** [comment]
Tested - works fine.

**Pull request merged**